### PR TITLE
WFLY-10084 Proxy Protocol

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: undertow/WFLY-10084_Proxy_Protocol.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+=== Affected Projects or Components:
+
+* Undertow
+
+== Requirements
+
+This will be implemented as an attribute called `proxy-protocol` under each of the listeners. If this is set to true
+then the listener will be wrapped by the Undertow proxy protocol implementation. This wrapper reads the proxy protocol
+header from the stream, before handing off the connection to the normal listener implementation.
```
Comment (bstansberry) [inline undertow/WFLY-10084_Proxy_Protocol.adoc]: What does it do with the information?  IOW how is supporting this protocol going to help solve concrete problems?  The ones I've heard about relate to making the original client address available to things like access loggers and calls to Keycloak.

I think I know in a vague hand-wavy kind of way. :)

**Comment by bstansberry** [comment]
Is there anything relevant to say about the undertow as a LB use case? The main scenario for this is for a server running behind HAProxy, and that doesn't seem to be about the LB use case. But people will try and use things in other ways.

**Comment by bstansberry** [comment]
@rcernich FYI.  Documents like this one is how we do initial requirement analysis for WF features, so if you have any feedback, comments here are a good place for it.

File: undertow/WFLY-10084_Proxy_Protocol.adoc (hunk 1 above)
Comment (rcernich) [reply undertow/WFLY-10084_Proxy_Protocol.adoc]: Ditto @bstansberry's question, the driving requirement is from access loggers, keycloak, etc.

File: undertow/WFLY-10084_Proxy_Protocol.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+* {issue-base-url}/UNDERTOW-573[UNDERTOW-573]
+* {issue-base-url}/EAP7-974[EAP7-974]
+
+=== Dev Contacts:
+
+* mailto:{email}[{author}]
+
+=== QE Contacts:
+
```
Comment (jstourac) [inline undertow/WFLY-10084_Proxy_Protocol.adoc]: feel free to put me in - jstourac, Jan Stourac

File: undertow/WFLY-10084_Proxy_Protocol.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+* {issue-base-url}/WFLY-10084[WFLY-10084]
+
+=== Related Issues:
+
+* {issue-base-url}/UNDERTOW-573[UNDERTOW-573]
+* {issue-base-url}/EAP7-974[EAP7-974]
+
+=== Dev Contacts:
+
+* mailto:jstourac@redhat.com[Jan Stourac]
```
Comment (jstourac) [inline undertow/WFLY-10084_Proxy_Protocol.adoc]: I've just noticed that I am mentioned in a wrong section :)

**Comment by jamezp** [comment]
This should have been merged for WildFly 13.

**Pull request merged**